plugins/net_plugin/include/eos/net_plugin/protocol.hpp

```diff
@@ -11,6 +11,9 @@ namespace eos {
       int16_t         network_version = 0;
       chain_id_type   chain_id; ///< used to identify chain
       fc::sha256      node_id; ///< used to identify peers and prevent self-connect
+      Name            producer; ///< first producer name on this node
+      fc::sha256      token; ///< digest to prove we own the private key of the producer named above
```

Comment:
This system is subject to replay attack, you must perform a ECDH key exchange and sign the shared secret.